### PR TITLE
fix: re evaluate tokens when they are registered in transformer registry

### PR DESCRIPTION
**Pull request by 0xged**

### CODE REVIEW

File: src/utils/token.ts [hunk 1]
```diff
@@ -70,18 +70,6 @@ export function getOrCreate(tokenAddress: Address, allowed: boolean): Token {
...
-      // allowed the token first insted of registering the transformer, so
-      // we need to re-evaluate if it's being allowed again 😵
-      if (allowed) {
-        const tokenTypeAndTransformerAddress = getTokenTypeAndTransformerAddress(tokenAddress);
-        token.type = tokenTypeAndTransformerAddress.tokenType;
-
-        if (tokenTypeAndTransformerAddress.tokenType != 'BASE') {
-          token.underlyingTokens = getUnderlyingTokenIds(tokenTypeAndTransformerAddress.transformerAddress, tokenAddress);
-        }
-      }
```
Comment (0xged) [inline src/utils/token.ts:84]: There is no longer a need for this. Since the 4626 registration would happen before this.

**Review by 0xsambugs — APPROVED** [review]
Looks good, just two small questions

File: src/utils/constants.ts [hunk 2]
```diff
@@ -10,3 +11,12 @@ export const ZERO_BD = BigDecimal.fromString('0');
 export const ONE_BD = BigDecimal.fromString('1');
 export const BI_18 = BigInt.fromI32(18);
 export const MAX_BI = BigInt.fromString('115792089237316195423570985008687907853269984665640564039457584007913129639935');
+
+export function getKnown4626(): Map<string, Array<Address>> {
+  const KNOWN_4626 = new Map<string, Array<Address>>();
+  KNOWN_4626.set('mainnet', new Array<Address>());
+  KNOWN_4626.set('optimism', new Array<Address>());
```
[the remark this answers is not in the document]
Comment (0xged) [reply src/utils/constants.ts:18]: image
Looks like it doesn't. I'll leave the query in the comment too:

```{
  position(id: 25220) {
    id
    from {
      name
    }
    to {
      name
    }
    toWithdrawUnderlyingAccum
    totalSwappedUnderlyingAccum
    totalWithdrawnUnderlying
    history {
      action
      ...on SwappedAction {
        swapped
        swappedUnderlying
      }
    }
  }
}```

File: src/mappings/transformerRegistry.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import * as transactionLibrary from '../utils/transaction';
+import * as tokenLibrary from '../utils/token';
+import { Hub } from '../../generated/Hub/Hub';
+import { TransformersRegistered } from '../../generated/Hub/TransformerRegistry';
+import { HUB_ADDRESS } from '../utils/constants';
+import { getTokenTypeAndTransformerAddress, getUnderlyingTokenIds } from '../utils/token';
+
+export function handleTransformersRegistered(event: TransformersRegistered): void {
+  log.info('[TransformerRegistry] Handle transformers registered', []);
+  transactionLibrary.getOrCreateFromEvent(event, 'Tr-TransformersRegistered');
```
[the remark this answers is not in the document]
Comment (0xged) [reply src/mappings/transformerRegistry.ts:11]: It saves the transaction in which the event happened. Hash, block number, etc. Although we are not using it to link anything, it has helped in the past to understand some timelines.

**Pull request merged**